lib/modules/@apostrophecms/ui/src/apos/components/AposAreaMenu.vue

```diff
@@ -0,0 +1,295 @@
+                >
+                  <AposAreaMenuItem
+                    @click="clicked(child)"
+                    :item="child"
+                    :tabbable="index === active"
+                    @up="switchItem(`child-${index}-${childIndex - 1}`, -1)"
+                    @down="switchItem(`child-${index}-${childIndex + 1}`, 1)"
+                  />
+                </li>
+              </ul>
+            </dd>
+          </dl>
+          <AposAreaMenuItem
+            v-else
+            @click="clicked(item)"
+            :item="item"
+            @up="switchItem(`item-${index - 1}`, -1)"
+            @down="switchItem(`item-${index + 1}`, 1)"
+          />
+        </li>
+      </ul>
+    </AposContextMenu>
+  </div>
+</template>
+
+<script>
+
+export default {
+  props: {
+    menu: {
+      type: Array,
+      required: true
+    }
+  },
+  emits: ['click'],
+  data() {
+    return {
+      active: 0,
+      groupIsFocused: false
+    };
+  },
+  computed: {
+    groupedMenus() {
+      let flag = false;
+      this.menu.forEach((e) => {
+        if (e.items) {
+          flag = true;
+        }
+      });
+      return flag;
+    },
+    myMenu() {
+      if (this.groupedMenus) {
+        return this.composeGroups();
+      } else {
+        return this.menu;
+      }
+    },
+    menuId() {
+      return `areaMenu-${(Math.floor(Math.random() * Math.floor(10000)))}`;
+    }
+  },
+  methods: {
+    groupFocused() {
+      this.groupIsFocused = true;
+    },
+    groupBlurred() {
+      this.groupIsFocused = false;
+    },
+    clicked(item) {
+      this.$emit('click', item.action);
+    },
+    composeGroups() {
+      const ungrouped = {
+        label: 'Ungrouped Widgets',
+        items: []
+      };
+      const myMenu = [];
+
+      this.menu.forEach((item) => {
+        if (!item.items) {
+          ungrouped.items.push(item);
+        } else {
+          myMenu.push(item);
+        }
+      });
+
+      if (ungrouped.items.length) {
+        myMenu.push(ungrouped);
+      }
+      return myMenu;
+    },
+
+    toggleGroup(index) {
+      if (this.active !== index) {
+        this.active = index;
+      } else {
+        this.active = null;
+      }
+    },
+
+    switchGroup(index, dir) {
+      let target;
+
+      if (dir > 0) {
+        target = index < this.$refs.groupButton.length - 1 ? index + 1 : 0;
+      }
+
+      if (dir < 0) {
+        target = index === 0 ? this.$refs.groupButton.length - 1 : index - 1;
+      }
+
+      if (dir === 0) {
+        target = 0;
+      }
+
+      if (!dir) {
+        target = this.$refs.groupButton.length - 1;
+      }
+
+      this.$nextTick(() => {
+        this.$refs.groupButton[target].focus();
+      });
+    },
+
+    switchItem(name, dir) {
+      if (this.$refs[name]) {
+        this.$refs[name][0].querySelector('button').focus();
+      }
+    }
+  }
+};
+</script>
+
+<style lang="scss" scoped>
+// @import '../../scss/_mixins';
```

Comment:
This should be able to be safely removed.

Reply:
+1